### PR TITLE
Add an example showing alternate mouse cursors.

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This makes it easy to test that cursors are working in a new backend (or if we add new cursors).

I'm not sure if it belongs in the `widgets` section.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: examples/widgets/mouse_cursor.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+"""
+============
+Mouse Cursor
+============
+
+This example sets the cursor on the canvas allowing you to see what they would
+look like. Note, this is an interactive example, and must be run to see the
```
Comment (jklymak) [inline examples/widgets/mouse_cursor.py:7]:
```suggestion
This example sets an alternative cursor on a figure canvas.  
Note, this is an interactive example, and must be run to see the
```

**Comment by timhoffm** [comment]
If we are ok with a row-only layout, we can simplify the code considerably: 

```
import matplotlib.pyplot as plt
from matplotlib.backend_tools import Cursors


fig, axs = plt.subplots(len(Cursors), gridspec_kw={'hspace': 0})
fig.suptitle('Hover over an Axes to see alternate Cursors')

for ax, cursor in zip(axs.flat, Cursors):
    ax.cursor_to_use = cursor
    ax.text(0.5, 0.5, cursor.name,
            horizontalalignment='center', verticalalignment='center')
    ax.set(xticks=[], yticks=[])


def hover(event):
    if fig.canvas.widgetlock.locked():
        # Don't do anything if the zoom/pan tools have been enabled.
        return

    fig.canvas.set_cursor(
        event.inaxes.cursor_to_use if event.inaxes else Cursors.POINTER)


fig.canvas.mpl_connect('motion_notify_event', hover)

plt.show()
```
grafik

**Comment by QuLogic** [comment]
It's just a ploy for that alternative layout generator that people wanted :stuck_out_tongue: But more seriously, I think the square aspect is a better mouse target than the wide and short ones. And if we add more cursors, it might get even more cramped.

**Comment by timhoffm** [comment]
The question is what the goal of the example is.

Is this about the code that is needed to change mouse cursors? Or is this only about showing the cursors? If the former, less distracting code is better. If the latter, I'm tempted to say that an example is too much effort for this. Users need to copy the code and run it. Showing some screenshots might be a simpler solution.

**Comment by QuLogic** [comment]
I'd rather not screenshots; cursors are backend dependent, and I'm not going to take (# backends)*(# cursors) screenshots. And everything in `widgets` needs to be run to see what they do, so I don't think that's much of an argument against the example. But if simplifying gets it in, I'll do that.

**Review by timhoffm — APPROVED** [review]

**Pull request merged**